c_secure_comm.c

```diff
@@ -477,28 +624,52 @@ session_key_list_t *send_session_key_req_via_TCP(SST_ctx_t *ctx) {
                 printf("Auth nonce verified!\n");
             }
             close(sock);
+            state = FINISHED;
             return session_key_list;
+        } else if (message_type == AUTH_ALERT) {
+            session_key_list->num_key = 0;
+            switch (data_buf[0]) {
+                case INVALID_DISTRIBUTION_KEY:
+                    printf("Error: Invalid Distribution Key\n");
+                    break;
+                case INVALID_SESSION_KEY_REQ:
+                    printf("Error: Invalid Session Key Request\n");
+                    break;
+                case UNKNOWN_INTERNAL_ERROR:
+                    printf("Error: Unknown Internal Error\n");
+                    break;
+                default:
+                    printf("Error: Unknown Code\n");
+                    break;
+            }
+            return NULL;
         }
     }
+    // Should not come here.
+    return NULL;
 }
 
-session_key_list_t *send_session_key_req_via_UDP(SST_ctx_t *ctx) {
-    session_key_list_t *s_key_list;
-    return s_key_list;
-    // TODO:(Dongha Kim) Implement this function.
-    error_exit("This function is not implemented yet.");
-}
+// session_key_list_t *send_session_key_req_via_UDP(SST_ctx_t *ctx) {
+//     session_key_list_t *s_key_list;
+//     return s_key_list;
+//     // TODO:(Dongha Kim) Implement this function.
+//     error_exit("This function is not implemented yet.");
+// }
 
 unsigned char *check_handshake1_send_handshake2(
     unsigned char *received_buf, unsigned int received_buf_length,
     unsigned char *server_nonce, session_key_t *s_key,
     unsigned int *ret_length) {
     unsigned int decrypted_length;
-    unsigned char *decrypted = symmetric_decrypt_authenticate(
-        received_buf + SESSION_KEY_ID_SIZE,
-        received_buf_length - SESSION_KEY_ID_SIZE, s_key->mac_key, MAC_KEY_SIZE,
-        s_key->cipher_key, CIPHER_KEY_SIZE, AES_CBC_128_IV_SIZE,
-        &decrypted_length);
+    unsigned char *decrypted;
+    if (symmetric_decrypt_authenticate(
+            received_buf + SESSION_KEY_ID_SIZE,
+            received_buf_length - SESSION_KEY_ID_SIZE, s_key->mac_key,
+            MAC_KEY_SIZE, s_key->cipher_key, CIPHER_KEY_SIZE,
+            AES_128_CBC_IV_SIZE, s_key->enc_mode, 0, &decrypted,
```

Comment:
Please use #define or enum for no HMAC mode instead of constant 0.

Reply:
Added here. 5f4ad9fd07423104f08fd26946d0c8bfe39aee11.

Reply:
Additional minor fix here.  37a349d32f38b8655ad8ebb29d4176e1a9a473fc